presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -541,6 +543,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            Optional<Expression> argumentNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
```

Comment:
> `select 1 as x from tpch.sf1.nation group by x`

This fails because there is no `x` column in `tpch.sf1.nation`. Group by expressions can reference only columns from `FROM` query. This should work:
```
select 1 as x from tpch.sf1.nation group by nation
```

Still I think you should fix this issue in this PR to make the grouping feature complete. This is a relatively small change and I don't think it's worth making another PR out of this issue. It can be fixed as: https://github.com/prestodb/presto/pull/7712#discussion_r111554472